File: Java/1_BinaryGap.java
// you can also use imports, for example:
// import java.util.*;

// https://app.codility.com/demo/results/training867KYX-BXR/

class Solution {
    public int solution(int N) {
        int mask = 1, maximum = 0, counter = 0;
        while(N != 0){
            if(N%2 == 0){ // reached 0
                if (counter > 0) counter++; // case: counting
                if (counter == -1) counter = 1;
            } 
            else { // reached 1
                if (counter == 0) counter=-1;
                else {
                    maximum = maximum>counter? maximum : counter;
                    counter = -1;
                } 
            }
            N = N>>1;
        }
        return maximum;
    }
}

File: C#/1. BinaryGap.cs
using System;
// you can also use other imports, for example:
// using System.Collections.Generic;

// you can write to stdout for debugging purposes, e.g.
// Console.WriteLine("this is a debug message");

class Solution {
    public int solution(int N) {
        // Implement your solution here
        int maxima = 0, mask = 1, zeroCounter = 0, start = 0;
        while(N!=0){
            if((N & mask) != 0){
                if (start==1) maxima = Math.Max(maxima, zeroCounter);
                start = 1;
                zeroCounter = 0;
            }
            else
                zeroCounter++;
            N = N>>1;
        }
        return maxima;
    }
}
